packages/cli/src/commands/task.ts

```diff
@@ -177,6 +283,16 @@ export const startDockerExecution = async (
     return;
   }
 
+  let isDockerRootless = false;
+
+  const dockerInfo = launchSync('docker', ['info', '-f', 'json']).stdout;
+  if (dockerInfo) {
+    const info = JSON.parse(dockerInfo.toString());
+    isDockerRootless = (info?.SecurityOptions || []).some((value: string) =>
+      value.includes('rootless')
+    );
+  }
```

Comment:
We should wrap this code in a `try/catch` block to capture errors with `launchSync` and JSON parse.